### ISSUE TITLE
qtox window too big and not scalable

### ISSUE THREAD

**Issue by ogbaba** [qTox/qTox#3637]
##### Brief Description

OS: Arch GNU/Linux
qTox version: 1.5.1
Commit hash:  ?
toxcore:  it says 0.0.0 weirdly
Qt:  5.7.0
Hardware:  Thinkpad x260 with 1920*1080 resolution 12.5 inches display

Reproducible: Always
##### Steps to reproduce
1. Start qtox
##### Observed Behavior

The qtox window is almost the size of the screen and can't be scaled down, it's quite uncomfortable
##### Expected Behavior

Smaller and scalable window 
##### Additional Info

![qtoxtoobig](https://cloud.githubusercontent.com/assets/13153084/17775579/207a59ec-6559-11e6-9c5b-c654a6462d58.png)

[log.txt](https://github.com/qTox/qTox/files/425031/log.txt)

**Comment by sudden6** [comment]
A possible workaround could be to use `QT_SCALE_FACTOR` environment variable as described in http://doc.qt.io/qt-5/highdpi.html altough this would make everything smaller and isn't equivalent to proper resizing.

**Comment by initramfs** [comment]
Do you have the variable `QT_DEVICE_PIXEL_RATIO` defined anywhere (in your environment)? That's an legecy variable to control Qt scaling that has since been deprecated.

Additionally, do you have `QT_AUTO_SCREEN_SCALE_FACTOR` defined? Qt only recently started supporting HiDPI screens, and it is required for that variable to be defined for "proper" scaling to take place (though, it shouldn't case a scaling that's "too big"). It may be possible that the DPI of your screen is stuck inbetween two scaling values (I believe the auto scaling factor sticks to integers. i.e. 2x, 3x, 4x, etc) such that it chooses a larger than optimal scaling value. If this is the case use `QT_SCALE_FACTOR` as @sudden6 describes.

**Comment by Dreadeath** [comment]
I confirm this in Arch Linux, other qt apps have proper scale. Though some time ago it was the other way, all qt apps were overscaled while qtox was "normal". Oh, and I have no tweaks in QT_\* env variables, if I apply e.g. QT_SCALE_FACTOR=0.5, graphics became normal, while fonts are very small.

**Comment by initramfs** [comment]
Have you tried defining `QT_AUTO_SCREEN_SCALE_FACTOR`?

**Comment by ogbaba** [comment]
I tried defining it but it didn't change anything

**Comment by phaoost** [comment]
I have the same issue on my system (Devuan ceres), however setting QT_AUTO_SCREEN_SCALE_FACTOR to 0 has fixed that

**Comment by phaoost** [comment]
Er, just have noticed that chats messages have too small font, increasing in user interface does not affect chat font

**Comment by ogbaba** [comment]
yes in fact it works I just didn't start it from terminal but then I have the same issue as @phaoost

**Comment by nazar-pc** [comment]
I think latest or one of latest nightly builds fixed this. Can anyone confirm this?

**Comment by cjsthompson** [comment]
This is still a problem for me (Arch Linux). qTox is the only Qt app I use that has this scaling problem.

![screenshot_2017-01-27_15-57-04](https://cloud.githubusercontent.com/assets/359383/22375181/54f88cbe-e4a9-11e6-92d0-2769f3b7c56c.png)

**Comment by ogbaba** [comment]
Same here, the issue is still present under Fedora 25

**Comment by FoundOne** [comment]
I'm on the latest git version under Arch Linux. All looks fine. The screen resolution is 1600x900.

**Comment by sudden6** [comment]
@ogbaba @cjsthompson have you tried the `QT_AUTO_SCREEN_SCALE_FACTOR`and `QT_SCALE_FACTOR` variables?

**Comment by zetok** [comment]
No response for over a month, assuming it was fixed, thus closing.

If issue is still not fixed, please comment for reopen.

**Comment by toxuser001001** [comment]
This is still a major issue, seemingly random across Debian / Mint / Ubuntu. Some installations have the window take up 75% of the screen, and some work perfectly.

**Comment by cjsthompson** [comment]
Actually, it's not a problem with Qtox. At first I only had this annoying problem with Qtox. But then I realized it happens randomly with many other Qt apps too. After investigating this, I found out it boils down to buggy monitor EDID firmware (Samsung) giving the wrong monitor size when plugged with HDMI. The Linux kernel has quirks for some monitors with EDID problems and X.org could too. I reported the problem on the X and Linux kernel bug trackers and seemingly (as that was over a year ago) no one there gives a shit about a problem with an older monitor since they're all running the latest and greatest 32 core machines and what their employers asks of them is their priority. Samsung are also notorious scum who won't release firmware updates. Fortunately, the EDID problem only happens with HDMI and not with DVI and VGA. So I just plugged the monitor with DVI and the problem is solved (at least if one doesn't need HDMI audio).
